std/algebra/twistededwards/bandersnatch/point.go

```diff
@@ -98,41 +107,87 @@ func (p *Point) Double(api frontend.API, p1 *Point, curve EdCurve) *Point {
 	return p
 }
 
+// phi endomorphism √-2 ∈ 𝒪₋₈
+// (x,y) → λ × (x,y) s.t. λ² = -2 mod Order
+func (p *Point) phi(api frontend.API, p1 *Point, curve EdCurve) *Point {
+
+	xy := api.Mul(p1.X, p1.Y)
+	yy := api.Mul(p1.Y, p1.Y)
+	f := api.Sub(1, yy)
+	f = api.Mul(f, curve.endo1)
+	g := api.Add(yy, curve.endo0)
+	g = api.Mul(g, curve.endo0)
+	h := api.Sub(yy, curve.endo0)
+
+	p.X = api.DivUnchecked(f, xy)
+	p.Y = api.DivUnchecked(g, h)
+
+	return p
+}
+
+var scalarDecompositionHint = hint.NewStaticHint(func(curve ecc.ID, inputs []*big.Int, res []*big.Int) error {
+	// TODO: handle properly negative scalars
```

Comment:
+ check curveID is what is expected

Reply:
at which level should it be checked?

Reply:
inside the hint I think, particularly if this hint works only for one curve; "if curve != ecc.BLS... return errors.New("....")"